vyos/vyos-build
# frr: T4737: Replaced patch for connected routes processing

## Change Summary

Replaced patch for connected routes processing

## Types of changes


## Related Task(s)

* https://vyos.dev/T4737

## Component(s) name

frr/zebra

## Proposed changes

An old patch breaks connected routes in a situation when more than one IP address is presented on an interface and it switches state from down to up.

A new version includes more backports from frr upstream, which work well regardless of an interface state.

## How to test

Case 1:

```
ip a flush dev eth0
ip link set eth0 up
ip a add 192.0.2.1/24 dev eth0 & ip a add 192.0.2.2/24 dev eth0
vtysh -c 'show ip route'
```

Case 2:

```
ip a flush dev eth0
ip link set eth0 down
ip a add 192.0.2.1/24 dev eth0
ip a add 192.0.2.2/24 dev eth0
ip link set eth0 up
vtysh -c 'show ip route'
```

In both cases FRR should see a single correct route:

```
Codes: K - kernel route, C - connected, S - static, R - RIP,
       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
       T - Table, v - VNC, V - VNC-Direct, A - Babel, D - SHARP,
       F - PBR, f - OpenFabric,
       > - selected route, * - FIB route, q - queued, r - rejected, b - backup

C>* 192.0.2.0/24 is directly connected, eth0, 00:00:01
```
## Checklist:
